bluepyentity/register.py

```
@@ -0,0 +1,472 @@
+                else:
+                    resource = self._forge.attach(path)
+                ret.append(resource)
+            return ret
+
+        return None
+
+    def _is_equal_to_nexus_resource(self, resource):
+        """Implements checking if the definition is equal to a nexus resource."""
+        return getattr(self.resource, "id", None) == resource.id
+
+    def _with_defaults(self, definition):
+        """Add default values to definiton."""
+        defaults = get_default_params(self.type)
+        return dict(defaults, **definition)
+
+    def _wrap_fields_with_ids(self, definition):
+        """Wrap attributes that contain IDs into objects."""
+        patch = {}
+        for field in self.possible_ids:
+            value = definition.get(field)
+            if isinstance(value, (str, list)):
+                patch[field] = _wrap_id_fetch(value, self._forge)
+
+        return dict(definition, **patch)
+
+    # TODO:
+    # - check if the definition conforms to expected schema/format
+    # --- if from workflow, can we just expect it to?
+    # --- how to activate this from nexus backend side or do we do it here?
+
+
+class DetailedCircuit(Resource):
+    """Class to register resources of type DetailedCircuit."""
+
+    required = {"name", "description", "circuitConfigPath", "circuitType"}
+
+    def _format_attributes(self, definition):
+        patch = {}
+        path = definition["circuitConfigPath"]
+
+        # NOTE: Does it need to be DataDownload, can it be just a URL as a string?
+        if isinstance(path, str):
+            patch["circuitConfigPath"] = {
+                "type": "DataDownload",
+                "url": _wrap_linked_file_path(path),
+            }
+
+        # NOTE: perhaps better to do this lookup when creating the resource
+        # Define brainLocation as suggested in DKE-1062
+        if brain_region := definition.get("brainRegion"):
+            brain_region = self._forge.retrieve(brain_region, cross_bucket=True)
```

imo all forge retrievals/registrations should be wrapped to check that the resource has been retrieved/registered successfully and raise appropriately to avoid silent error propagation...